src/utils/node.ts

```diff
@@ -0,0 +1,21 @@
+interface ICoords {
+  readonly x: number;
+  readonly y: number;
+}
+
+export function isNodeOverlapped(node: HTMLElement, rect: DOMRect): boolean {
+  const center = findRectCenterCoords(rect);
+  const topElement = document.elementFromPoint(center.x, center.y);
+  return !node.isSameNode(topElement) && !node.contains(topElement);
+}
+
+export function findRectCenterCoords(rect: DOMRect): ICoords {
+  const x1 = rect.x;
+  const x2 = rect.x + rect.width;
```

Comment:
`DOMRect` has `right` and `bottom`, you know ... and this whole function and interface is totally unnecessary, as it doesn't need to be exported (not used from outside)